test-resources/infrastructure/public-api.yaml

```diff
@@ -14,6 +14,18 @@ paths:
           description: authorizationCode generated by CRI
           schema:
             type: string
+        - name: client_id
+          in: query
+          required: false
```

Comment:
If core behaves like this, why isn't this `required: true`

Reply:
I think it is 
https://github.com/govuk-one-login/ipv-core-back/blob/main/lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/ProcessCriCallbackHandler.java#L154

then `input body` of the core-callback Request is https://github.com/govuk-one-login/ipv-core-back/blob/main/libs/common-services/src/main/java/uk/gov/di/ipv/core/library/dto/CriCallbackRequest.java#L19

We are interested in:
    private String authorizationCode;
    private String credentialIssuerId;
    private String redirectUri;

I think `credentialIssuerId` is an identifier that can represent the audience, given that this is an `internal api` i think we can just send the audience and the redirect url as query params instead of using state

Reply:
So ipv-core frontend corresponds as https://github.com/govuk-one-login/ipv-core-front/blob/main/src/app/credential-issuer/middleware.ts#L32-L46

Reply:
I'm wondering if it's this middleware from Core that we should be looking at actually https://github.com/govuk-one-login/ipv-core-front/blob/main/src/app/oauth2/middleware.ts